source/slang/check.cpp

```diff
@@ -1927,10 +1927,16 @@ namespace Slang
         // (via the given `inheritanceDecl`) actually provides
         // members to satisfy all the requirements in the interface.
         bool checkInterfaceConformance(
+            HashSet<DeclRef<InterfaceDecl>> & checkedInterfaceDeclRef,
             DeclRef<AggTypeDeclBase>        typeDeclRef,
             InheritanceDecl*    inheritanceDecl,
             DeclRef<InterfaceDecl>  interfaceDeclRef)
         {
+            if (!checkedInterfaceDeclRef.Contains(interfaceDeclRef))
```

Comment:
Hmm... Related to my comment above, it looks like you are already trying to avoid duplicate checking here.